src/combinations.rs

```diff
@@ -54,7 +54,7 @@ impl<I: Iterator> Combinations<I> {
 
     /// Returns a reference to the source iterator.
     #[inline]
-    pub(crate) fn src(&self) -> &I { &self.pool.it }
+    pub(crate) fn src(&self) -> &Fuse<I> { &self.pool.it }
```

Comment:
This function is only used in `Powerset::size_hint`, where we ask for `src().size_hint` - I think that using the size_hint of the fused iterator is actually what we want there. Can you confirm this?

I guess that `Powerset::size_hint` should - in the future - also redirect to `LazyBuffer::size_hint`, no? Then we could get rid of this `src` exposing the innards of `LazyBuffer` entirely

Reply:
Yes entirely, I confirm that `Powerset::size_hint` wants the not yet written `LazyBuffer::size_hint` from the wrapped `Combinations`.
`fn src` might become `fn pool(&self) -> &LazyBuffer<I>` later in order for `Powerset::size_hint` to `let src_total = self.combs.pool().size_hint();` instead of doing the job of `LazyBuffer::size_hint`.